iarc_main/scripts/follow.py

```diff
@@ -24,9 +25,20 @@ def run(self):
 
 	            # Plan the motion
 	            cmd = Twist()
-	            cmd.linear.x = position[0]
-	            cmd.linear.y = position[1]
-	            cmd.angular.z = 0
+
+	            # maxvelocity of our drone
+	            maxvelocity = 1
+
+	            # diagonalvelocity = the velocity we would get if we use position[0] and position[1]
+	            diagonalvelocity = math.sqrt(position[0]**2 + position[1]**2)
+
+	            # Scale down our velocity_x and velocity_y so that our diagonalvelocity won't exceed maxvelocity
+	            # But if diagonalvelocity is already < max_velocity then use diagonalvelocity instead (slow down when approach nearer)
+	            cmd.linear.x = position[0] if diagonalvelocity < maxvelocity else position[0] / diagonalvelocity * maxvelocity
+	            cmd.linear.y = position[1] if diagonalvelocity < maxvelocity else position[1] / diagonalvelocity * maxvelocity
```

Comment:
It may be more readable to write these lines as

    if diagonalvelocity < maxvelocity:
        cmd.linear.x=...
        cmd.linear.y=...
    else:
        ...


I think it will make it easier to see that there are exactly two behaviors, and generally improve readability.

Feel free to disagree, this is just my 2 cents.

Reply:
I agree. Thanks for suggestion!